lstchain/hdu/hdu_table.py

```diff
@@ -0,0 +1,464 @@
+            t_aeff['HDU_NAME'] = ['EFFECTIVE AREA']
+
+            hdu_tables.append(t_aeff)
+        else:
+            print('Effective Area HDU not found')
+
+        ###############################################
+
+        # Filling up the remainng quantities
+        obs_id.append(event_table.meta['OBS_ID'])
+        source.append(event_table.meta['OBJECT'])
+        N_TELS.append(event_table.meta["N_TELS"])
+        TELLIST.append(event_table.meta["TELLIST"])
+
+        ontime.append(event_table.meta["ONTIME"])
+        livetime.append(event_table.meta["LIVETIME"])
+        deadc.append(event_table.meta["DEADC"])
+        tstart.append(gti_table["START"])
+        tstop.append(gti_table["STOP"])
+
+        ra_pnt.append(pointing_table.meta['RA_PNT'])
+        dec_pnt.append(pointing_table.meta['DEC_PNT'])
+        zen_pnt.append(90 - float(pointing_table.meta['ALT_PNT']))
+        alt_pnt.append(pointing_table.meta['ALT_PNT'])
+        az_pnt.append(pointing_table.meta['AZ_PNT'])
+
+
+    #Complete HDU table
+    hdu_table = vstack(hdu_tables)
+    hdu_table.meta['EXTNAME'] = 'HDU_INDEX'
+    hdu_table.meta["HDUCLASS"] = "GADF"
+    hdu_table.meta["HDUDOC"] = "https://github.com/open-gamma-ray-astro/gamma-astro-data-formats"
+    hdu_table.meta["HDUVERS"] = "0.2"
+    hdu_table.meta["HDUCLAS1"] = "INDEX"
+    hdu_table.meta["HDUCLAS2"] = "HDU"
+    hdu_table.meta["TELESCOP"] = "CTA"
+    hdu_table.meta["INSTRUME"] = "LST-1"
+
+    filename_hdu_table = 'hdu-index.fits.gz'
+    hdu = fits.BinTableHDU(hdu_table)
+    hdu_name='HDU_INDEX'
+    hdu_list = fits.HDUList()
+    hdu_list.append(hdu)
+    #fits.append(fits_dir+filename_hdu_table, hdu_table, header='HDU_INDEX')
+    hdu_list.writeto(str(fits_dir)+"/"+filename_hdu_table, overwrite=True)
+
+    #Complete OBS table
+    obs_table = Table(
+        [obs_id, ra_pnt, dec_pnt, zen_pnt, alt_pnt, az_pnt, ontime, livetime, deadc, tstart, tstop, source, N_TELS, TELLIST],
+        names=(
+            'OBS_ID', 'RA_PNT', 'DEC_PNT', 'ZEN_PNT', 'ALT_PNT', 'AZ_PNT', 'ONTIME', 'LIVETIME', 'DEADC', 'TSTART',
+            'TSTOP', 'OBJECT','N_TELS', 'TELLIST'),
+        dtype=('>i8', '>f4', '>f4', '>f4', '>f4', '>f4', '>f4', '>f4', '>f4', '>f8', '>f8', 'S20','>i8', 'S20'),
+        meta={'name': 'OBS_INDEX'}
+    )
+
+    obs_table = vstack(obs_table)
+    obs_table.meta['EXTNAME'] = 'OBS_INDEX'
+    obs_table.meta["HDUCLASS"] = "GADF"
+    obs_table.meta["HDUDOC"] = "https://github.com/open-gamma-ray-astro/gamma-astro-data-formats"
+    obs_table.meta["HDUVERS"] = "0.2"
+    obs_table.meta["HDUCLAS1"] = "INDEX"
+    obs_table.meta["HDUCLAS2"] = "OBS"
+    obs_table.meta["TELESCOP"] = "CTA"
+    obs_table.meta["INSTRUME"] = "LST-1"
+
+    obs_table.meta['MJDREFI'] = event_table.meta['MJDREFI']
+    obs_table.meta['MJDREFF'] = event_table.meta['MJDREFF']
+
+    filename_obs_table = 'obs-index.fits.gz'
+    obs = fits.BinTableHDU(obs_table)
+    hdu_list = fits.HDUList()
+    hdu_list.append(obs)
```

Comment:
You need a primary HDU as the first hdu, just do `hdu_list= fits.HDUList([fits.PrimaryHDU] + obs)`

Reply:
Understood